spec-main/chromium-spec.ts

```diff
@@ -973,117 +973,25 @@ describe('chromium features', () => {
-        done()
-      })
-      w.webContents.on('page-title-updated', () => {
-        const parsedURL = url.parse(w.webContents.getURL(), true)
-        expect(parsedURL.protocol).to.equal('chrome:')
-        expect(parsedURL.hostname).to.equal('pdf-viewer')
-        expect(parsedURL.query.src).to.equal(pdfSource)
-        expect(w.webContents.getTitle()).to.equal('cat.pdf')
-      })
-      w.webContents.loadURL(pdfSource)
-    })
-
-    it('opens a pdf link given params, the query string should be escaped', (done) => {
-      const w = createBrowserWindow({ plugins: true, preload: 'preload-pdf-loaded.js' })
-      ipcMain.once('pdf-loaded', (event, state) => {
-        expect(state).to.equal('success')
-        done()
-      })
-      w.webContents.on('page-title-updated', () => {
-        const parsedURL = url.parse(w.webContents.getURL(), true)
-        expect(parsedURL.protocol).to.equal('chrome:')
-        expect(parsedURL.hostname).to.equal('pdf-viewer')
-        expect(parsedURL.query.src).to.equal(pdfSourceWithParams)
-        expect(parsedURL.query.b).to.be.undefined()
-        expect(parsedURL.search!.endsWith('%3Fa%3D1%26b%3D2')).to.be.true()
-        expect(w.webContents.getTitle()).to.equal('cat.pdf')
-      })
-      w.webContents.loadURL(pdfSourceWithParams)
-    })
-
-    it('should download a pdf when plugins are disabled', async () => {
-      const w = createBrowserWindow({ plugins: false, preload: 'preload-pdf-loaded.js' })
-      w.webContents.loadURL(pdfSource)
-      const [state, filename, mimeType] = await new Promise(resolve => {
-        session.defaultSession.once('will-download', (event, item) => {
-          item.setSavePath(path.join(fixturesPath, 'mock.pdf'))
-          item.on('done', (e, state) => {
-            resolve([state, item.getFilename(), item.getMimeType()])
-          })
-        })
-      })
-      expect(state).to.equal('completed')
-      expect(filename).to.equal('cat.pdf')
-      expect(mimeType).to.equal('application/pdf')
-      fs.unlinkSync(path.join(fixturesPath, 'mock.pdf'))
-    })
-
-    it('should not open when pdf is requested as sub resource', async () => {
+    it('opens when loading a pdf resource as top level navigation', async () => {
       const w = new BrowserWindow({ show: false })
-      w.loadURL('about:blank')
-      const [status, title] = await w.webContents.executeJavaScript(`fetch(${JSON.stringify(pdfSource)}).then(res => [res.status, document.title])`)
-      expect(status).to.equal(200)
-      expect(title).to.not.equal('cat.pdf')
-    })
-
-    it('opens when loading a pdf resource in a iframe', (done) => {
-      testPDFIsLoadedInSubFrame('pdf-in-iframe.html', 'preload-pdf-loaded-in-subframe.js', done)
+      w.loadURL(pdfSource)
+      const [, contents] = await emittedOnce(app, 'web-contents-created')
+      expect(contents.getURL()).to.equal('chrome-extension://mhjfbmdgcfjbbpaeojofohoefgiehjai/index.html')
+      await emittedOnce(contents, 'did-finish-load')
```

Comment:
Is there a better signal or chrome api we can run to see if the pdf is actually loaded , maybe https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/resources/pdf/pdf_viewer.js;bpv=0;bpt=0;l=1521 ? `did-finish-load` on the webcontents wouldn't exactly denote that the extension rendered the pdf.

Reply:
i tried a lot of things to try to get a better "loaded" signal but nothing i tried worked. the core issue is that the MimeHandlerGuestView is similar to a <webview>, and it doesn't forward any kind of messages out of the <embed>. the best idea i have is to make preloads work in extension pages and use that to add some hooks, but we don't have that capability yet.

Reply:
I'm currently having some issue loading [larger PDF file](https://www.hq.nasa.gov/alsj/a17/A17_FlightPlan.pdf). Is this somehow related?

Reply:
@ericshung no, I don't think that's related, but I'm seeing some other issues with that PDF (including discovering a bug in Chromium itself: https://crbug.com/1051548). Thanks for the report.